src/main/java/org/hydev/mcpm/client/installer/InstallInteractor.java

```diff
@@ -46,20 +57,25 @@ public InstallInteractor(PluginDownloader spigotPluginDownloader, DatabaseManage
      * @param installInput: the plugin submitted by the user for installing
      * */
     @Override
-    public InstallResult installPlugin(InstallInput installInput)
+    public boolean installPlugin(InstallInput installInput,
+                                       ResultPresenter resultPresenter)
```

Comment:
In clean architecture, your presenter should depend on the use case, not the other way around

Reply:
Wait why is this me I don't remember changing this. Isn't this from the installer PR?

Reply:
oops, thank you! I will fix it by passing the presenter to the interactor constructor instead of use case method.

Reply:
what I mean is that like before, you should return an InstallResult so the result can go the the controller, who passes it to the presenter

Reply:
Ohh, I am actually thinking of changing the InstallResult returned type to boolean, letting the presenter handle the job showing the state of manual install input and its dependency. If I return the InstallInput to the controller, I won't be able to know what happened to the dependency, whether or not they have been installed. I think this way works well for testing too ><